### PR TITLE
setup ossf scorecard and codql workflows 

### PR DESCRIPTION
**Pull request by mmorel-35**
This also fixes dependabot config as npm is used while pip is not

Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ updates:
   directory: /
   schedule:
     interval: daily
-- package-ecosystem: pip
```
Comment (phlax) [inline .github/dependabot.yml:7]: do we not still need this ?

Comment (mmorel-35) [reply .github/dependabot.yml:7]: There is no docs folder in the project so dependabot doesn’t identify any dependency to update.

**Review by phlax — DISMISSED** [review]
lgtm, thanks @mmorel-35

**Comment by mmorel-35** [comment]
Anything left here ?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  push:
+    branches:
+    - "main"
+  pull_request:
+    branches:
+    - "main"
+  schedule:
+  - cron: '16 11 * * 5'
+
+permissions:
```
Comment (phlax) [inline .github/workflows/codeql.yml:13]: lets shift this to be consistent now the others are all updated

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  contents: read
+
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-22.04' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
+    permissions:
```
Comment (phlax) [inline .github/workflows/codeql.yml:22]: ditto this

File: .github/workflows/scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+name: Scorecard supply-chain security
+on:
+  branch_protection_rule:
+  schedule:
+  - cron: '33 13 * * 5'
+  push:
+    branches:
+    - "main"
+
+permissions:
```
Comment (phlax) [inline .github/workflows/scorecard.yml:10]: and these 2

**Review by phlax — COMMENTED** [review]
i should have posted as review - but just a few small nits and then gtm

thanks again @mmorel-35 !!!

**Review by phlax — APPROVED** [review]
lgtm, thanks

separately we should bump the ubuntu version i reckon

**Comment by mmorel-35** [comment]
@phlax ,
This seems ready to go

**Review by phlax — APPROVED** [review]
lgtm, thanks @mmorel-35

**Pull request merged**